reputation/reputation.go

```diff
@@ -0,0 +1,576 @@
+func createNamedRandRows(names []string) []NodeReputationRecord {
+	var res []NodeReputationRecord
+
+	for idx, name := range names {
+		res = append(res, createNamedRow(idx, name))
+	}
+
+	return res
+}
+
+// insertRows inserts the slice of reputation row structs based on the insert string
+func insertRows(db *sql.DB, rows []NodeReputationRecord, insertString string) error {
+	tx, err := db.Begin()
+	defer tx.Rollback()
+	if err != nil {
+		log.Printf("%q: %s\n", err, insertString)
+		return InsertError.Wrap(err)
+	}
+
+	insertStmt, err := tx.Prepare(insertString)
+	if err != nil {
+		log.Printf("%q: %s\n", err, insertString)
+		return InsertError.Wrap(err)
+	}
+	defer insertStmt.Close()
+
+	for _, row := range rows {
+		_, err = insertStmt.Exec(
+			row.name,
+			row.uptime,
+			row.auditSuccess,
+			row.auditFail,
+			row.latency,
+			row.amountOfDataStored,
+			row.falseClaims,
+			row.shardsModified,
+		)
+		if err != nil {
+			log.Printf("%q: %s\n", err, insertString)
+			return InsertError.Wrap(err)
+		}
+	}
+	tx.Commit()
+
+	return nil
+}
+
+// selectFromDB side effect function that prints the rows from the query string
+func selectFromDB(db *sql.DB, selectString string) error {
+	rows, err := db.Query(selectString)
+	if err != nil {
+		log.Printf("%q: %s\n", err, selectString)
+		return SelectError.Wrap(err)
+	}
+	defer rows.Close()
+
+	transformedRows, err := iterOnDBRows(rows)
+	if err != nil {
+		log.Printf("%q: %s\n", err, selectString)
+		return SelectError.Wrap(err)
+	}
+
+	for _, row := range transformedRows {
+		// side effect
+		fmt.Println(row)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		log.Printf("%q: %s\n", err, selectString)
+		return SelectError.Wrap(err)
+	}
+
+	return nil
+}
+
+// iterOnDBRows iterate on rows in the database to transform into slice of NodeReputationRecord
+func iterOnDBRows(rows *sql.Rows) ([]NodeReputationRecord, error) {
+	var res []NodeReputationRecord
+
+	for rows.Next() {
+		var row NodeReputationRecord
+
+		err := rows.Scan(
+			&row.name,
+			&row.timestamp,
+			&row.uptime,
+			&row.auditSuccess,
+			&row.auditFail,
+			&row.latency,
+			&row.amountOfDataStored,
+			&row.falseClaims,
+			&row.shardsModified,
+		)
+		if err != nil {
+			log.Printf("%q\n", err)
+			return nil, IterError.Wrap(err)
+		}
+
+		res = append(res, row)
+	}
+
+	return res, nil
+}
+
+// getNodeReputationRecords function that returns a slice of reputation rows based on the query string
+func getNodeReputationRecords(db *sql.DB, selectString string) ([]NodeReputationRecord, error) {
+	rows, err := db.Query(selectString)
+	if err != nil {
+		log.Printf("%q: %s\n", err, selectString)
+		return nil, SelectError.Wrap(err)
+	}
+	defer rows.Close()
+
+	res, err := iterOnDBRows(rows)
+	if err != nil {
+		log.Printf("%q: %s\n", err, selectString)
+		return nil, SelectError.Wrap(err)
+	}
+
+	err = rows.Err()
+	if err != nil {
+		log.Printf("%q: %s\n", err, selectString)
+		return nil, SelectError.Wrap(err)
+	}
+
+	return res, nil
```

Comment:
some comments for the rest of these functions - no need to log here, just return the error, and yeah, from line 217 on, you can just return rows.Err() and avoid the rest